api/middleware/authMiddleware.js

```
@@ -0,0 +1,1008 @@
+  }
+  
+  // Additional security checks
+  // Prevent injection patterns and malicious inputs
+  const maliciousPatterns = [
+    /[<>\"\'`]/,        // HTML/JS injection characters
+    /\$\{/,             // Template literal injection
+    /\.\./,             // Path traversal
+    /[;|&]/,            // Command injection
+    /javascript:/i,     // Protocol injection
+    /data:/i,           // Data URL injection
+    /\x00/,             // Null byte injection
```

Biome (1.9.4)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)